src/main/java/org/owasp/wrongsecrets/Challenges.java

```diff
@@ -0,0 +1,94 @@
+package org.owasp.wrongsecrets;
+
+import static java.util.stream.Collectors.toMap;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import lombok.Getter;
+import org.owasp.wrongsecrets.challenges.Challenge;
+import org.owasp.wrongsecrets.definitions.ChallengeDefinition;
+import org.owasp.wrongsecrets.definitions.ChallengeDefinitionsConfiguration;
+import org.owasp.wrongsecrets.definitions.Difficulty;
+import org.owasp.wrongsecrets.definitions.Navigation;
+
+/**
+ * A collection of all challenges. This class glues a challenge definition together with the
+ * challenge class.
+ */
+public class Challenges {
+
+  @Getter private final ChallengeDefinitionsConfiguration definitions;
+  private final Map<String, ChallengeDefinition> urlNameToDefinition;
+  private final Map<String, Challenge> classNameToChallenge;
+  private final Map<ChallengeDefinition, List<Challenge>> challengeDefinitionToChallenge;
+
+  public Challenges(ChallengeDefinitionsConfiguration definitions, List<Challenge> challenges) {
+    this.definitions = definitions;
+
+    urlNameToDefinition =
+        definitions.challenges().stream()
+            .collect(toMap(definition -> definition.name().url(), Function.identity()));
+    classNameToChallenge =
+        challenges.stream()
+            .collect(toMap(challenge -> challenge.getClass().getName(), Function.identity()));
+    challengeDefinitionToChallenge =
+        definitions.challenges().stream()
+            .collect(
+                toMap(
+                    definition -> definition,
+                    definition ->
+                        definition.sources().stream()
+                            .map(source -> classNameToChallenge.get(source.className()))
+                            .toList()));
+  }
+
+  public Navigation navigation(ChallengeDefinition challengeDefinition) {
+    return new Navigation(definitions.challenges(), challengeDefinition);
+  }
+
+  public int numberOfChallenges() {
+    return definitions.challenges().size();
+  }
+
+  public Optional<ChallengeDefinition> findByShortName(String shortName) {
```

Comment:
can we do by description as well and expose to the FE as a search?

Reply:
This should be `findByUrl` I will update this

Reply:
Can we park this as a future improvement? At the search box does work

Reply:
yes we can :D , i think having jsut the table search is fine actually.